test/cypress/e2e/multi-datasources/multi_datasources_enabled.spec.js

```diff
@@ -0,0 +1,244 @@
+        title: externalTitle,
+        endpoint: 'https://localhost:9202',
+        auth: {
+          type: 'username_password',
+          credentials: {
+            username: 'admin',
+            password: 'myStrongPassword123!',
+          },
+        },
+      },
+    },
+  });
+};
+
+const deleteAllDataSources = () => {
+  cy.request(
+    'GET',
+    `${Cypress.config(
+      'baseUrl'
+    )}/api/saved_objects/_find?fields=id&fields=description&fields=title&per_page=10000&type=data-source`
+  ).then((resp) => {
+    if (resp && resp.body && resp.body.saved_objects) {
+      resp.body.saved_objects.map(({ id }) => {
+        cy.request({
+          method: 'DELETE',
+          url: `${Cypress.config('baseUrl')}/api/saved_objects/data-source/${id}`,
+          body: { force: false },
+          headers: {
+            'osd-xsrf': true,
+          },
+        });
+      });
+    }
+  });
+};
+
+const createUrlParam = (label, id) => {
+  const dataSourceObj = { label, id };
+  return `?dataSource=${JSON.stringify(dataSourceObj).toString()}`;
+};
+
+let externalDataSourceId;
+let externalDataSourceUrl;
+let localDataSourceUrl;
+
+describe('Multi-datasources enabled', () => {
+  beforeEach(() => {
+    localStorage.setItem('opendistro::security::tenant::saved', '""');
+    localStorage.setItem('home:newThemeModal:show', 'false');
+    createDataSource().then((resp) => {
+      if (resp && resp.body) {
+        externalDataSourceId = resp.body.id;
+      }
+      externalDataSourceUrl = createUrlParam(externalTitle, externalDataSourceId);
+      localDataSourceUrl = createUrlParam('Local cluster', '');
+    });
+  });
+
+  afterEach(() => {
+    cy.clearCookies();
+    cy.clearAllLocalStorage();
+    cy.clearAllSessionStorage();
+    deleteAllDataSources();
+  });
+
+  it('Checks Get Started Tab', () => {
+    // Remote cluster purge cache
+    cy.visit(
+      `http://localhost:5601/app/security-dashboards-plugin${externalDataSourceUrl}#/getstarted`
+    );
+
+    cy.contains('h1', 'Get started');
+    cy.get('[data-test-subj="dataSourceSelectableContextMenuHeaderLink"]').should(
+      'contain',
+      '9202'
+    );
+
+    cy.get('[data-test-subj="purge-cache"]').click();
+    cy.get('[class="euiToast euiToast--success euiGlobalToastListItem"]')
+      .get('.euiToastHeader__title')
+      .should('contain', 'successful for 9202');
+  });
+
+  it('Checks Auth Tab', () => {
+    cy.visit(`http://localhost:5601/app/security-dashboards-plugin${externalDataSourceUrl}#/auth`);
+
+    cy.get('.panel-header-count').first().invoke('text').should('contain', '(1)');
+  });
+
+  it('Checks Users Tab', () => {
+    cy.request({
+      method: 'POST',
+      url: `http://localhost:5601/api/v1/configuration/internalusers/9202-user?dataSourceId=${externalDataSourceId}`,
+      headers: {
+        'osd-xsrf': true,
+      },
+      body: {
+        backend_roles: [''],
+        attributes: {},
+        password: 'myStrongPassword123!',
```

Comment:
Distinct user should have distinct password to prevent unaligned auth scenarios

Reply:
The user is only used to see if they show up in the external datasource user list, but that is a good point, so I changed the password.